pyccel/parser/syntactic.py

```diff
@@ -210,8 +209,8 @@ def _visit(self, stmt):
             return result
 
         # Unknown object, we raise an error.
-        errors.report(PYCCEL_RESTRICTION_UNSUPPORTED_SYNTAX, symbol=stmt,
-                      severity='fatal')
+        return errors.report(PYCCEL_RESTRICTION_UNSUPPORTED_SYNTAX, symbol=stmt,
+                      severity='error')
```

Comment:
I find it a bit strange to call `errors.report` within a `return` statement. I suppose that `errors.report == None`, but `return None` is automatically added at the end of every Python function...

Reply:
This usually happens with `severity='fatal'` (so the return doesn't actually happen) to avoid codacy's errors about [inconsistent-return-statements](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/inconsistent-return-statements.html).

This shouldn't be used here. Fixed 721fce9d